lib/gooddata/models/domain.rb

```diff
@@ -220,25 +221,29 @@ def users(domain, id = :all, opts = {})
         client = client(opts)
         domain = client.domain(domain)
         if id == :all
-          GoodData.logger.warn("Retrieving all users from domain #{domain.name}")
-          all_users = []
-          page_limit = opts[:page_limit] || 1000
-          offset = opts[:offset] || 0
-          loop do
-            begin
-              tmp = client(opts).get("#{domain.uri}/users", params: { offset: offset, limit: page_limit })
+          if @all_domain_users[domain.name].nil?
```

Comment:
Do not cache it here, it will effect all usecases need domain users. Cache it in user filters only.